src/app/content/components/Page/highlightManager.ts

```diff
@@ -163,10 +193,34 @@ export default (container: HTMLElement, getProp: () => HighlightProp) => {
         ;
 
       highlighter.clearFocus();
-      const focusedId = getProp().focused;
-      const focused = focusedId && highlighter.getHighlight(focusedId);
-      if (focused) {
-        focused.focus();
+      const { scrollTarget, focus, focused: focusedId, highlightsLoaded, loggedOut, page } = getProp();
+      const focused = focusedId ? highlighter.getHighlight(focusedId) : null;
+      const stateEstablished = (highlightsLoaded || (loggedOut && page));
+
+      const highlightScrollTarget = scrollTarget && isHighlightScrollTarget(scrollTarget) ? scrollTarget : null;
+      const scrollTargetHighlight = highlightScrollTarget && highlighter.getHighlight(highlightScrollTarget.id);
+
+      const toFocus = getHighlightToFocus(
+        focused, prevProps.focused, pendingHighlight, scrollTargetHighlight, scrollTargetHighlightIdThatWasHandled);
+      if (toFocus) {
+        toFocus.focus();
+
+        if (options) {
+          options.onSelect(toFocus);
+        }
+
+        if (
+          scrollTargetHighlight
+          && toFocus.id === scrollTargetHighlight.id
+          && toFocus.id !== focusedId
+          && toFocus.id !== scrollTargetHighlightIdThatWasHandled) {
+          focus(toFocus.id);
+          (toFocus.elements[0] as HTMLElement).scrollIntoView();
+          scrollTargetHighlightIdThatWasHandled = scrollTargetHighlight.id;
```

Comment:
I think it would be nice to have this logic in a separate function close to `setPendingHighlight`. I feel like there is a lot of noise in `manager.update`

Reply:
Is this not resolved? Did you mean something different than I did?